src/bodywork/workflow_execution.py

```diff
@@ -167,24 +152,32 @@ def run_workflow(
         )
         _log.error(msg)
         try:
-            if config.project.run_on_failure and type(e) not in [
-                BodyworkNamespaceError,
-                BodyworkDockerImageError,
-                BodyworkGitError,
-            ]:
+            if (
+                config is not None
+                and config.project.run_on_failure
+                and type(e)
+                not in [
+                    BodyworkNamespaceError,
+                    BodyworkDockerImageError,
+                    BodyworkGitError,
+                    BodyworkConfigError,
+                ]
+            ):
                 _run_failure_stage(
                     config, e, namespace, repo_url, repo_branch, docker_image
                 )
         except Exception as ex:
             failure_msg = (
-                f"Error executing failure stage: {config.project.run_on_failure}"
+                f"Error executing failure stage: {config.project.run_on_failure}"   # type: ignore # noqa
```

Comment:
Why do you need to ignore the type - this is usually a bad smell?

Reply:
See comment above.

Reply:
This and your following comments are all related to MyPy errors. The config can't logically be null but the MyPy check can't calculate it.